### ISSUE TITLE
Fails node.gyp rebuild with current Jessie

### ISSUE THREAD

**Issue by Shane-Lester** [marvinroger/node-rcswitch#6]
I have a fresh install of Jessie on a Pi B and I can't get the npm package to install. It fails node.gyp rebuild. I have updated npm and can install other npm packages. Any help would be appreciated.

**Comment by marvinroger** [comment]
Could you please send the output of `npm install rcswitch`?
Le 20 janv. 2016 3:18 PM, "Shane Lester" notifications@github.com a
écrit :

**Comment by Shane-Lester** [comment]
```
pi@raspberrypi:~/Documents/node_projects $ npm install rcswitch

> rcswitch@0.3.0 install /home/pi/Documents/node_projects/node_modules/rcswitch
> node-gyp rebuild

make: Entering directory '/home/pi/Documents/node_projects/node_modules/rcswitch               /build'
  CXX(target) Release/obj.target/rcswitch/src/rcswitch.o
  CXX(target) Release/obj.target/rcswitch/src/RCSwitchNode.o
../src/RCSwitchNode.cpp: In static member function ‘static void RCSwitchNode::Sw               itchOp(const Nan::FunctionCallbackInfo<v8::Value>&, bool)’:
../src/RCSwitchNode.cpp:49:35: error: no matching function for call to ‘Nan::Utf               8String::Utf8String(v8::Handle<v8::Value>&)’
       Nan::Utf8String sGroup(group);
                                   ^
../src/RCSwitchNode.cpp:49:35: note: candidates are:
In file included from ../src/RCSwitchNode.h:4:0,
                 from ../src/RCSwitchNode.cpp:1:
../../nan/nan.h:1243:33: note: Nan::Utf8String::Utf8String(const Nan::Utf8String               &)
   NAN_DISALLOW_ASSIGN_COPY_MOVE(Utf8String)
                                 ^
../../nan/nan.h:111:35: note: in definition of macro ‘NAN_DISALLOW_COPY’
 # define NAN_DISALLOW_COPY(CLASS) CLASS(const CLASS&);
                                   ^
../../nan/nan.h:1243:3: note: in expansion of macro ‘NAN_DISALLOW_ASSIGN_COPY_MO               VE’
   NAN_DISALLOW_ASSIGN_COPY_MOVE(Utf8String)
   ^
../../nan/nan.h:1243:33: note:   no known conversion for argument 1 from ‘v8::Ha               ndle<v8::Value>’ to ‘const Nan::Utf8String&’
   NAN_DISALLOW_ASSIGN_COPY_MOVE(Utf8String)
                                 ^
../../nan/nan.h:111:35: note: in definition of macro ‘NAN_DISALLOW_COPY’
 # define NAN_DISALLOW_COPY(CLASS) CLASS(const CLASS&);
                                   ^
../../nan/nan.h:1243:3: note: in expansion of macro ‘NAN_DISALLOW_ASSIGN_COPY_MO               VE’
   NAN_DISALLOW_ASSIGN_COPY_MOVE(Utf8String)
   ^
In file included from ../src/RCSwitchNode.h:4:0,
                 from ../src/RCSwitchNode.cpp:1:
../../nan/nan.h:1210:23: note: Nan::Utf8String::Utf8String(v8::Local<v8::Value>)
   NAN_INLINE explicit Utf8String(v8::Local<v8::Value> from) :
                       ^
../../nan/nan.h:1210:23: note:   no known conversion for argument 1 from ‘v8::Ha               ndle<v8::Value>’ to ‘v8::Local<v8::Value>’
../src/RCSwitchNode.cpp:62:34: error: no matching function for call to ‘Nan::Utf               8String::Utf8String(v8::Handle<v8::Value>&)’
       Nan::Utf8String v8str(famly);
                                  ^
../src/RCSwitchNode.cpp:62:34: note: candidates are:
In file included from ../src/RCSwitchNode.h:4:0,
                 from ../src/RCSwitchNode.cpp:1:
../../nan/nan.h:1243:33: note: Nan::Utf8String::Utf8String(const Nan::Utf8String               &)
   NAN_DISALLOW_ASSIGN_COPY_MOVE(Utf8String)
                                 ^
../../nan/nan.h:111:35: note: in definition of macro ‘NAN_DISALLOW_COPY’
 # define NAN_DISALLOW_COPY(CLASS) CLASS(const CLASS&);
                                   ^
../../nan/nan.h:1243:3: note: in expansion of macro ‘NAN_DISALLOW_ASSIGN_COPY_MO               VE’
   NAN_DISALLOW_ASSIGN_COPY_MOVE(Utf8String)
   ^
../../nan/nan.h:1243:33: note:   no known conversion for argument 1 from ‘v8::Ha               ndle<v8::Value>’ to ‘const Nan::Utf8String&’
   NAN_DISALLOW_ASSIGN_COPY_MOVE(Utf8String)
                                 ^
../../nan/nan.h:111:35: note: in definition of macro ‘NAN_DISALLOW_COPY’
 # define NAN_DISALLOW_COPY(CLASS) CLASS(const CLASS&);
                                   ^
../../nan/nan.h:1243:3: note: in expansion of macro ‘NAN_DISALLOW_ASSIGN_COPY_MO               VE’
   NAN_DISALLOW_ASSIGN_COPY_MOVE(Utf8String)
   ^
In file included from ../src/RCSwitchNode.h:4:0,
                 from ../src/RCSwitchNode.cpp:1:
../../nan/nan.h:1210:23: note: Nan::Utf8String::Utf8String(v8::Local<v8::Value>)
   NAN_INLINE explicit Utf8String(v8::Local<v8::Value> from) :
                       ^
../../nan/nan.h:1210:23: note:   no known conversion for argument 1 from ‘v8::Ha               ndle<v8::Value>’ to ‘v8::Local<v8::Value>’
rcswitch.target.mk:90: recipe for target 'Release/obj.target/rcswitch/src/RCSwit               chNode.o' failed
make: *** [Release/obj.target/rcswitch/src/RCSwitchNode.o] Error 1
make: Leaving directory '/home/pi/Documents/node_projects/node_modules/rcswitch/               build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_               modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:809:               12)
gyp ERR! System Linux 4.1.13+
gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/b               in/node-gyp.js" "rebuild"
gyp ERR! cwd /home/pi/Documents/node_projects/node_modules/rcswitch
gyp ERR! node -v v0.10.29
gyp ERR! node-gyp -v v3.2.1
gyp ERR! not ok
/home/pi/Documents/node_projects
└── piswitch@1.2.3

npm WARN enoent ENOENT, open '/home/pi/Documents/node_projects/package.json'
npm WARN node_projects No description
npm WARN node_projects No repository field.
npm WARN node_projects No README data
npm WARN node_projects No license field.
npm ERR! Linux 4.1.13+
npm ERR! argv "node" "/usr/local/bin/npm" "install" "rcswitch"
npm ERR! node v0.10.29
npm ERR! npm  v3.5.3
npm ERR! code ELIFECYCLE

npm ERR! rcswitch@0.3.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the rcswitch@0.3.0 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the rcswitch package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs rcswitch
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls rcswitch
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/pi/Documents/node_projects/npm-debug.log
```

Do you want the npm-debug.log file?

Thanks

Shane

**Comment by marvinroger** [comment]
No thanks, the log is complete enough. I guess you're using Node.js >= 4.0.0? Which version exactly?

**Comment by marvinroger** [comment]
Sorry, you're using node v0.10.29! I'll take a look. ;)

**Comment by marvinroger** [comment]
I am sorry I still haven't had the time to look into it. Hopefully this week.

**Comment by Shane-Lester** [comment]
Not a problem. I appreciate whatever time you can spare. 

Shane

**Comment by marvinroger** [comment]
Could you please try with the latest git version?

```
git clone https://github.com/marvinroger/node-rcswitch.git
cd node-rcswitch
npm install --unsafe-perm
```

**Comment by Shane-Lester** [comment]
Builds without issue now. 

Thanks

Shane

**Comment by Shane-Lester** [comment]
Builds perfectly now. Thanks

**Comment by marvinroger** [comment]
Great, I just published the fix on npm (0.3.1).
